### PR TITLE
Adding test to trigger problem reported in #2916

### PR DESCRIPTION
**Pull request by hkaiser**
- flyby: channel::close() now returns the number of elements left in the channel

### CODE REVIEW

**Comment by biddisco** [comment]
This branch somehow got the debugging commit on this branch that was changed to use the HAVE_UNISTD_H check instead. You should just remove that commit and rebase onto the later master.

**Comment by hkaiser** [comment]
Will do, @sithhell added this here for some reason.

File: tests/regressions/lcos/channel_2916.cpp [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        hpx::shared_future<int> result = target.then(
+            [&count](hpx::shared_future<locality_id_t> t)
+            ->  hpx::shared_future<int>
+            {
+                --count;
+                return hpx::make_ready_future(0);
+            });
+
+        // Free resources
+        result.then([&free_resources, target, &count](hpx::shared_future<int>)
```
Comment (sithhell) [inline tests/regressions/lcos/channel_2916.cpp]: This will still fail as `free_resources` is captured by reference.

**Comment by hkaiser** [comment]
@sithhell your comments have been addressed.

**Pull request merged**